sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/InterpretedUnsafeProjection.scala

```diff
@@ -254,7 +254,7 @@ object InterpretedUnsafeProjection {
         (_, _) => {}
 
       case _ =>
-        throw QueryExecutionErrors.dataTypeUnsupportedError(dt)
+        throw QueryExecutionErrors.dataTypeUnsupportedForWriterFuncError(dt)
```

Comment:
seems we can never reach here unless we have a bug. maybe we don't need error class here and can throw `IllegalStateException` directly?

Reply:
How about to not handle the default case at all (I mean remove it). The compiler should output a warning if we forget to handle a type, and we.will get a match error exception in the case too.